lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/restart/DFSEvaluatorLogOverwriteReaderWriter.java

```diff
@@ -82,7 +82,7 @@ public synchronized void writeToEvaluatorLog(final String formattedEntry) throws
           inputStream = newEntryInputStream;
         }
 
-        IOUtils.copyBytes(inputStream, outputStream, 4096, true);
+        IOUtils.copyBytes(inputStream, outputStream, 4096, false);
       } finally {
```

Comment:
Do you mean something like 

    try (InputStream inputStream = null) {
        ...
    } finally {
        outputStream.hsync();
    }

It doesn't work, resources declared in `try` are final, so can't reassign `inputStream` later.